client/my-sites/current-site/index.jsx

```diff
@@ -159,9 +156,13 @@ const CurrentSite = React.createClass( {
 // TODO: make this pure when sites can be retrieved from the Redux state
 module.exports = connect(
 	( state, ownProps ) => {
+		const selectedSiteId = getSelectedSiteId( state );
 		const selectedSite = getSelectedSite( state );
+
 		return {
-			selectedSite: selectedSite || ownProps.sites.getPrimary()
+			selectedSiteId,
```

Comment:
Yeah, this is a good point.

>it leaves me wondering if getPrimary() has much value at all?

I wonder this as well. it appears `<CurrentSite />` is only used in the My Sites sidebar. When would it ever be the case that we'd want to defer to the primary site and not the "selected" site? Based on the [site selection middleware](https://github.com/Automattic/wp-calypso/blob/24f141b6c19ddc699ace7e81f3d08dc02d99e147/client/my-sites/controller.js#L113-L221), it seems the only case that we'd not have selected site exist is in the case of "All Sites" view, which doesn't need the `selectedSite` prop at all, as far as I can tell (unless we're relying on some strange behavior where we show domain warnings for primary site even in All Sites view).